### PR TITLE
Merge binascii module into 2and3

### PR DESCRIPTION
**Pull request by euresti**
Kept the python 3 version but modified it so that python 2 also accepts unicode strings.

### CODE REVIEW

File: stdlib/2and3/binascii.pyi [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    # Python 2 accepts unicode ascii pretty much everywhere. 
+    _Bytes = Union[bytes, Text]
+else:
+    # Python 3 only accepts bytes on the b2a methods.
+    _Bytes = bytes
+
+def a2b_uu(string: Union[Text, bytes]) -> bytes: ...
+def b2a_uu(data: _Bytes) -> bytes: ...
+def a2b_base64(string: Union[Text, bytes]) -> bytes: ...
+def b2a_base64(data: _Bytes) -> bytes: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/binascii.pyi]: This has another positional-only parameter as of Python 3: `    newline: bool(accept={int}) = True`.

File: stdlib/2and3/binascii.pyi [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+def a2b_qp(string: Union[Text, bytes], header: bool = ...) -> bytes: ...
+def b2a_qp(data: _Bytes, quotetabs: bool = ..., istext: bool = ...,
+             header: bool = ...) -> bytes: ...
+def a2b_hqx(string: Union[Text, bytes]) -> bytes: ...
+def rledecode_hqx(data: _Bytes) -> bytes: ...
+def rlecode_hqx(data: _Bytes) -> bytes: ...
+def b2a_hqx(data: _Bytes) -> bytes: ...
+def crc_hqx(data: _Bytes, crc: int) -> int: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/binascii.pyi]: The argument is called `oldcrc` in the Python 2 docstring, but it's positional-only so it doesn't really matter. If we want to be really precise we can make the arguments positional-only in Python 2 (and probably also pre-3.5 Python 3, but I haven't checked), but I think it's fine to ignore.

**Comment by JelleZijlstra** [comment]
Also, there's some trailing whitespace (see Travis) and I just noticed that the documentation says `Changed in version 3.3: ASCII-only unicode strings are now accepted by the a2b_* functions.`. We technically still support 3.2.

File: stdlib/2and3/binascii.pyi (hunk 1 above)
Comment (euresti) [inline stdlib/2and3/binascii.pyi]: Will add.

File: stdlib/2and3/binascii.pyi (hunk 2 above)
Comment (euresti) [inline stdlib/2and3/binascii.pyi]: It's positional only in both.  Technically this entire module is positonal only everywhere.

Comment (JelleZijlstra) [inline stdlib/2and3/binascii.pyi]: Not in 3.6. Presumably it was changed to use Argument Clinic at some point.

```In [5]: binascii.b2a_base64(data=b'')
Out[5]: b'\n'
```

Comment (euresti) [inline stdlib/2and3/binascii.pyi]: Oh interesting. Well crc is still positional only at least in 3.6.1

```
>>> binascii.crc32(b'foo', crc=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: crc32() takes no keyword arguments
```

File: stdlib/2and3/binascii.pyi [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import sys
+from typing import Union, Text
+
+
+if sys.version_info < (3,):
+    # Python 2 accepts unicode ascii pretty much everywhere.
+    _Bytes = Union[bytes, Text]
+    _Ascii = Union[bytes, Text]
+elif sys.version_info <= (3, 2):
```
Comment (JelleZijlstra) [inline stdlib/2and3/binascii.pyi]: should be `< (3, 3)`. But we're probably going to drop 3.2 support anyway (#1145).

**Pull request merged**